vaapi/vaapicontext.cpp

```diff
@@ -25,10 +25,63 @@
 #include "vaapi/vaapicontext.h"
 
 #include "common/log.h"
+#include "common/common_def.h"
 #include "vaapi/vaapidisplay.h"
 #include "vaapi/vaapiutils.h"
+#include <algorithm>
+#include <vector>
 
 namespace YamiMediaCodec{
+using std::vector;
+static const VAProfile h264ProfileList[] = {VAProfileH264ConstrainedBaseline, VAProfileH264Main, VAProfileH264High};
+
+//Driver may declare support higher profile but don't support lower profile.
+//In this case, higher profile should be created
+static bool checkH264Profile(VAProfile& profile, vector<VAProfile> profileList)
+{
+    vector<VAProfile>h264Profiles(h264ProfileList, h264ProfileList + N_ELEMENTS(h264ProfileList));
+    vector<VAProfile>::iterator start = std::find(h264Profiles.begin(), h264Profiles.end(), profile);
+    vector<VAProfile>::iterator result = std::find_first_of(profileList.begin(), profileList.end(), start, h264Profiles.end());
+
+    if (result != profileList.end()){
+        profile = *result;
+    } else{
+        // VAProfileH264Baseline is super profile for VAProfileH264ConstrainedBaseline
+        // old i965 driver incorrectly claims supporting VAProfileH264Baseline, but not VAProfileH264ConstrainedBaseline
+        if (profile == VAProfileH264ConstrainedBaseline && std::count(profileList.begin(), profileList.end(), VAProfileH264Baseline))
```

Comment:
I know find is more efficiency than count generally speaking. but still I use count instead of find in last verison, because profile list is short,  and using count need less code line, so there is no obvious efficiency gap.